wp-parsely.php

```diff
@@ -764,7 +764,33 @@ public function insert_parsely_page() {
 		// Assign default values for LD+JSON
 		// TODO: Maping of an install's post types to Parse.ly post types (namely page/post).
 		$parsely_page = $this->construct_parsely_metadata( $parsely_options, $post );
-		include 'parsely-parsely-page.php';
+
+		// Something went wrong - abort.
+		if ( empty( $parsely_page ) || ! isset( $parsely_page['headline'] ) ) {
+			return;
+		}
+
+		echo "\n" . '<!-- BEGIN Parse.ly ' . esc_html( Parsely::VERSION ) . ' -->' . "\n";
+
+		// Insert JSON-LD or repeated metas.
+		if ( 'json_ld' === $parsely_options['meta_type'] ) {
+			include __DIR__ . '/views/json-ld.php';
```

Comment:
```suggestion
			require __DIR__ . '/views/json-ld.php';
```

I usually tend toward `require` if it's a file that ships w/ my code that I expect to be present.  Is there a good reason to use `include` instead?

Reply:
`require` will cause a fatal if it's not present / accessible, while `include` just throws a warning. A lack of Parse.ly output (JSON-LD, repeated metas, custom meta), isn't a reason to kill the whole page/request IMO.

The files are expected to be present and accessible, but it's not critical to the success of the page.

Reply:
I generally prefer to fail by design when sourced plugin components are not present -- it can make diagnosing the issue substantially more difficult in practice

That said, I understand the point of not wanting to break the front-end of a site in such cases, so let's leave `include` in place for these.

I definitely think we should default to `require` for any admin views, however, or at least manually handle cases where `include`d files are not present instead of failing silently.

Reply:
Agreed for admin views - the purpose of viewing an admin page is to see/access the content of it, so a missing view file there _is_ a critical issue.